astropy/config/configuration.py

```diff
@@ -608,10 +608,14 @@ def get_config(packageormod=None, reload=False, rootname=None):
         try:
             # This feature is intended only for use by the unit tests
             if _override_config_file is not None:
-                cfgfn = _override_config_file
+                cfgfn = Path(_override_config_file)
             else:
-                cfgfn = path.join(get_config_dir(rootname=rootname), pkgname + ".cfg")
-            cobj = configobj.ConfigObj(cfgfn, interpolation=False)
+                cfgfn = (
+                    Path(get_config_dir(rootname=rootname))
+                    .joinpath(pkgname)
+                    .with_suffix(".cfg")
+                )
+            cobj = configobj.ConfigObj(str(cfgfn), interpolation=False)
```

Comment:
Also flagging for followup... maybe ConfigObj should primarily work with paths as well.

Reply:
This class lives in astropy.extern so we should probably not tinker with it.

Reply:
This seems like quite an undertaking, and I'm not sure that there would be a consensus on this as long as it's not broken. Would you like to open an issue for that ?

Reply:
https://github.com/astropy/astropy/issues/16476 ?